src/opm/parser/eclipse/EclipseState/Schedule/MSW/WellSegments.cpp

```diff
@@ -119,30 +104,25 @@ namespace Opm {
         // for the first record, which provides the information for the top segment
         // and information for the whole segment set
         const auto& record1 = welsegsKeyword.getRecord(0);
-        m_well_name = record1.getItem("WELL").getTrimmedString(0);
-
-        m_segments.clear();
-
         const double invalid_value = Segment::invalidValue(); // meaningless value to indicate unspecified values
 
-        m_depth_top = record1.getItem("DEPTH").getSIDouble(0);
-        m_length_top = record1.getItem("LENGTH").getSIDouble(0);
-        m_length_depth_type = LengthDepthFromString(record1.getItem("INFO_TYPE").getTrimmedString(0));
-        m_volume_top = record1.getItem("WELLBORE_VOLUME").getSIDouble(0);
+        double depth_top = record1.getItem("DEPTH").getSIDouble(0);
+        double length_top = record1.getItem("LENGTH").getSIDouble(0);
+        double volume_top = record1.getItem("WELLBORE_VOLUME").getSIDouble(0);
+        LengthDepth length_depth_type = LengthDepthFromString(record1.getItem("INFO_TYPE").getTrimmedString(0));
```

Comment:
For these, most likely should be `const`. If anything later modified them by accident, it might not be easy to debug.

Reply:
Fixed